File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    width = 0
    height = 0
    
    for i in sizes:    
      # 세로가 큰 것들은 돌려본다
      if i[1] > i[0]:
        width = max(width, i[1])
        height = max(height, i[0])
      else:
        width = max(width, i[0])
        height = max(height, i[1])
      
    return width * height

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

typedef vector<vector<int>> vvi;
int answer = 1e9;

int solution(vector<vector<int>> sizes) {
    int width = 0, height = 0;
    
    for (int i = 0; i < sizes.size(); i++) {
        if (sizes[i][0] > sizes[i][1]) {
            width = max(width, sizes[i][0]);
            height = max(height, sizes[i][1]);
        } else {
            width = max(width, sizes[i][1]);
            height = max(height, sizes[i][0]);
        }
    }
    return width * height;
}